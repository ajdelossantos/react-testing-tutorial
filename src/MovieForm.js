import React, { Component } from 'react'

export default class MovieForm extends Component {
  state = {
    text: '',
  };

  render() {
    const { submitForm } = this.props,
                    text = this.state;

    return (
      <div>
        <form
          data-testid="movie-form"
          onSubmit={ () => submitForm({ text }) }
        >
          <input type="text" />
          <button>
            Submit
          </button>
        </form>
      </div>
    );
  }
}
