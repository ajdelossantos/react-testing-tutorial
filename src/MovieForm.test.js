import React from 'react';
import {
  render,
  cleanup,
  fireEvent
} from 'react-testing-library';
import MovieForm from './MovieForm';





test('<MovieForm />', () => {
  afterEach(cleanup);

  const onSubmit = jest.fn(),
  {
        debug,
        getByTestId,
        container,
        getByText,
        getByLabelText
  }              = render(<MovieForm submitForm={ onSubmit } />);

    // debug();

    expect(getByTestId('movie-form')).toBeTruthy()

    // Alt syntax -- might not work
    // getByLabelText('Text').value = 'hello';
    // fireEvent.change(getByLabelText('Text'));

    fireEvent.change(getByLabelText('Text'), {
      target: { value: 'hello' }
    });

    fireEvent.click(getByText('Submit'));
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenCalledWith({ text: 'hello', });


    expect(container.firstChild).toMatchSnapshot();

});