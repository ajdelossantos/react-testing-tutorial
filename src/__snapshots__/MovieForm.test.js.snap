// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieForm /> 1`] = `
<div>
  <form
    data-testid="movie-form"
  >
    <label
      for="text"
    >
      Text
      <input
        type="text"
        id="text"
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</div>
`;
